src/KafkaConsumer/Processor/KafkaProcessor.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Confluent.Kafka;
+using KafkaConsumer.Processor.Config;
+using KafkaConsumer.TopicPartitionQueue;
+using Microsoft.Extensions.Options;
+
+namespace KafkaConsumer.Processor
+{
+	public class KafkaProcessor<TKey, TValue> : IKafkaProcessor<TKey, TValue>
+	{
+		private readonly IConsumer<TKey, TValue> _consumer;
+		private readonly ITopicPartitionQueueSelector<TKey, TValue> _topicPartitionQueueSelector;
+		private readonly ProcessorConfig _config;
+
+		public KafkaProcessor(
+			IConsumer<TKey, TValue> consumer,
+			ITopicPartitionQueueSelector<TKey, TValue> topicPartitionQueueSelector,
+			IOptions<ProcessorConfig> config)
+		{
+			_consumer = consumer;
+			_topicPartitionQueueSelector = topicPartitionQueueSelector;
+			_config = config.Value;
+		}
+
+		public async Task ProcessMessagesAsync(CancellationToken ct)
+		{
+			_consumer.Subscribe(_config.Topic);
+
+			try
+			{
+				while (!ct.IsCancellationRequested)
+				{
+					var consumeResult = Consume(ct);
+
+					if (consumeResult != null)
+					{
+						var queue = _topicPartitionQueueSelector.Select(consumeResult.TopicPartition);
+
+						var message = new MessageHandler.Message<TKey, TValue>(_consumer, consumeResult);
+
+						var isEnqueued = await queue.TryEnqueueAsync(message);
+
+						if (!isEnqueued)
+						{
+							await queue.AbortAsync();
+						}
+					}
+				}
+			}
+			catch (OperationCanceledException) { }
+			finally
+			{
+				_consumer.Close();
```

Comment:
Add a `TODO` about known issue, that `Close()` blocks indefinitely, if an exception was thrown from PartitionsAssigned/Revoked handler.